shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1831,6 +1834,7 @@ - (void)insertText:(NSString*)text {
   [copiedRects release];
   _selectionAffinity = _kTextAffinityDownstream;
   [self replaceRange:_selectedTextRange withText:text];
+  [self ensureUndoEnabled];
```

Comment:
Why do we only call this when insertText or delete text is called?

Reply:
I believe insert/delete are the only actions that should enable the undo buttons.

There is a strange issue with the undoManager where the first call to `- registerUndoWithTarget` doesn't actually enable the undo button. To work around this, this method calls `- registerUndoWithTarget`, then `- removeAllActionsWithTarget` to remove that placeholder. After that, calling `- registerUndoWithTarget` again will correctly enable the undo button.

Reply:
The framework could also perform actions in behalf of the user that are potentially undo/redo-able?